autolab/Autolab
# API endpoint for refreshing watchlist instances

## Description
This PR adds the API endpoint for bulk refreshing watchlist instances for a course. In addition, it also patched app/models/risk_conditions.rb such that the method update_current_for_course now utilizes transaction.

## Motivation and Context
This is part of the metrics feature functionality.

## How Has This Been Tested?
1. Add a new course with a reasonable number of grace days.
2. Add at least 3 users to that course.
3. Add a reasonable number of assignments, each consuming preferably different number of grace days (by tweaking due date and end date). For ease of later testing, set only one problem to each assignment with weight 100. Act as each user added in step 2. Submit as these users such that the number of no-submissions for each is varied. Act as instructor again and grade their submissions. Make sure the grades are spread over a reasonable range and pay attention to the ups and downs of one particular user. Take note of each user's trend so that comparison is easier for later purposes. After all of this is done, make sure to change the end date of each assignment to a past date. Unreleased grades are not considered for the grade-related conditions.
**Testing is first done case by case.**
4. Add a grace day usage condition to the course. Visit AutoPopulated/metrics/refresh_watch_list_instances. See if the returned instances match expectation. Tweak the condition to see if refreshed instances reflect the modification.
5. Add a grade drop condition. Visit AutoPopulated/metrics/refresh_watch_list_instances. See if the returned instances match expectation. Tweak the condition to see if refreshed instances reflect the modification.
6. Add a no submissions condition. Visit AutoPopulated/metrics/refresh_watch_list_instances. See if the returned instances match expectation. Tweak the condition to see if refreshed instances reflect the modification.
7. Add a low grades condition. Visit AutoPopulated/metrics/refresh_watch_list_instances. See if the returned instances match expectation. Tweak the condition to see if refreshed instances reflect the modification.
**Testing is then done in a hybrid manner.**
8. Update current conditions such that they are a hybrid of some of the four types. See if the refreshed instances match expectation.
**Lastly**
9. Refresh twice in a row. There shouldn't be any change in the records returned by the endpoint.
10. It should also be the case that none of the returned instances should include instructors' information.

## Types of changes


## Checklist:


## Other issues / help required




If unsure, feel free to submit first and we'll help you along.
